### PR TITLE
B 21000 int - Allow customer to edit and view Counseling Office

### PR DESCRIPTION
**Pull request by JonSpight**
## Summary
We want to give the customer the following capabilities:
Allow the customer to edit the counseling office.
Allow the customer to review the counseling office.

### How to test
**Create move**
1. Login as Customer
2. Create a move with Origin duty location as Scott AFB
3. Select a counseling office for the move and submit as normal

**Once move is created**
1. Select to view current move 
2. Verify the counseling office is displayed in the orders summary
3. select edit on the orders summary
7. change the counseling office and submit
8. verify the counseling office is displayed in the orders summary
9. selected edit on the order summary
10. change the origin duty location to NS Mayport
11. verify the counseling office is blank in the orders summary

## 508 Screenshots
[B-21000-508Screenshots.pdf](https://github.com/user-attachments/files/17263226/B-21000-508Screenshots.pdf)

### CODE REVIEW

**Comment by pambecker** [comment]
conflicts

File: pkg/handlers/internalapi/orders.go [hunk 1]
```diff
@@ -340,6 +341,37 @@ func (h UpdateOrdersHandler) Handle(params ordersop.UpdateOrdersParams) middlewa
 				}
 				order.OriginDutyLocation = &originDutyLocation
 				order.OriginDutyLocationID = &originDutyLocationID
+
+				if payload.MoveID != "" {
+					fmt.Print("Starting Counseling test")
+					moveID, err := uuid.FromString(payload.MoveID.String())
+					if err != nil {
+						fmt.Print("Jonathan 1 - Invalid move ID")
```
Comment (pambecker) [inline pkg/handlers/internalapi/orders.go:349]: Need to remove all these fmt.Print() lines in this class

File: swagger-def/internal.yaml [hunk 2]
```diff
@@ -200,7 +200,14 @@ definitions:
       mto_shipments:
         $ref: '#/definitions/MTOShipments'
       closeout_office:
-        $ref: 'definitions/TransportationOffice.yaml'
+        $ref: "definitions/TransportationOffice.yaml"
+      counseling_office:
+        $ref: "definitions/TransportationOffice.yaml"
+      counseling_office_id:
```
Comment (pambecker) [inline swagger-def/internal.yaml:206]: It seems redundant to have the id here too.  It's already in the TransportationOffice.yaml, can't we pull it from there to use everywhere?

File: swagger-def/internal.yaml [hunk 3]
```diff
@@ -2361,6 +2372,13 @@ definitions:
         $ref: '#/definitions/MTOShipments'
       closeoutOffice:
         $ref: 'definitions/TransportationOffice.yaml'
+      counselingOffice:
+        $ref: 'definitions/TransportationOffice.yaml'
+      counselingOfficeID:
```
Comment (pambecker) [inline swagger-def/internal.yaml:2377]: Same thing here, I don't think we want to have the ID here when we have it in the TransporationOffice.yaml already

File: pkg/handlers/internalapi/moves.go [hunk 4]
```diff
@@ -118,6 +121,12 @@ func payloadForInternalMove(storer storage.FileStorer, list models.Moves) []*int
 			currentMove.PrimeCounselingCompletedAt = *handlers.FmtDateTime(*move.PrimeCounselingCompletedAt)
 		}
 
+		var counselingOffice internalmessages.TransportationOffice
+		if move.CounselingOffice != nil {
```
Comment (paulstonebraker) [inline pkg/handlers/internalapi/moves.go:125]: can we skip creating the variable like you do in lines 73-75? `currentMove.CounselingOffice = payloads.TransportationOffice(*move.CounselingOffice)` would that work?

**Review by paulstonebraker — APPROVED** [review]
Everything works, made one small suggestion, but non-blocking

File: pkg/handlers/internalapi/orders.go [hunk 5]
```diff
@@ -340,6 +340,32 @@ func (h UpdateOrdersHandler) Handle(params ordersop.UpdateOrdersParams) middlewa
...
+				if payload.MoveID != "" {
+					moveID, err := uuid.FromString(payload.MoveID.String())
+					if err != nil {
+						return handlers.ResponseForError(appCtx.Logger(), err), err
+					}
+					move, err := models.FetchMove(appCtx.DB(), appCtx.Session(), moveID)
+					if err != nil {
+						return handlers.ResponseForError(appCtx.Logger(), err), err
+					}
+					if originDutyLocation.ProvidesServicesCounseling {
```
[the remark this answers is not in the document]
Comment (pambecker) [reply pkg/handlers/internalapi/orders.go:353]: Resolving this comment, Jon explained it to me, that even with Prime counseled as no, we'd still have to get move data and save to set the counseling office to null.  I'm good with how this is.

**Review by pambecker — APPROVED** [review]

**Pull request merged**